datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -166,8 +173,17 @@ private void populateDefaultProducerConfigs() {
 
   private Optional<Producer<K, V>> maybeGetKafkaProducer(DatastreamTask task) {
     Producer<K, V> producer = _kafkaProducer;
+    if (!_tasks.contains(task)) {
```

Comment:
Does access to _tasks not need to be synchronized?

Reply:
_tasks is a concurrent Hash map and based on its usage, it really does not have to be in synchronized block. We have added other checks as well to catch the misuse. for eg, every send checks if the task is present in _tasks or not.